### ISSUE TITLE
DSO203 HW2.60, Windows 10/11 DFU

### ISSUE THREAD

**Issue by kneave** [gabonator/LA104#113]
I'm trying to get my ancient DSO203 v2.6 working with LA104, it seems a wonderful upgrade for the scope! 

I've tried to update on my Windows 10 and 11 machines but the DFU drive doesn't appear, I've then (foolishly) tried hacking around with various DFU version using the internal UART connection with the obvious results of it not working either.

I found the original firmwares for everything on the web archive by the way, it's a great source it seems as Seeed's site has changed that much!
http://web.archive.org/web/20141110095005/http://www.seeedstudio.com/wiki/index.php?title=DSO_Quad:Upgrading_Firmware#Hardware_V2.6

Is there a known good image for the entire device I could upload via the UART connection that would get me running again please? I know you've mentioned dapboot on another issue and wondering if it works for the dso203 as well to get it working on 10/10?

Any tips would be gratefully received!

**Comment by gabonator** [comment]
For recovery you could flash the bin file in `/image` folder from this zip file: https://github.com/MotoMaxis/DS203-DSOQuad/blob/master/RECOVERY281282.zip

I was looking around if I could find full flash dump of ds203 with my fw, but no luck. I can prepare it for you, but it might take some time (a week?), so at first try to revert back to original fw and then try copying the new firmware over DFU drive (there is a trick, that you need to copy the .hex file twice, the second time it flashes it properly)

**Comment by MotoMaxis** [comment]
Greetings, 

Since he said the hardware of his DS203 is v2.6 the above package will not work as its for v2.81 or v2.82.  I 've seen an image for older hw somewhere but can't find it at the moment.

**Comment by kneave** [comment]
Thanks folks, I'm not in a massive hurry so if you could build an image that would be amazing! DFU 3.10 doesn't make it easy to do anything with a modern version of Windows, annoyingly I can't find any of my old ISOs for older Windows versions either. I even tried plugging in to my Android phone to see if it would show up there without any luck 😅

Thanks both for your hard work keeping this hardware alive by the way, improving it too, highly appreciated!

**Comment by kneave** [comment]
I don't even get the drive appearing in Windows 10/11 with DFU 3.10 by the way, I asked over on MiniDSO's forum if they had a newer version of DFU for 2.60 hardware but they just said it wasn't public. I'm not sure if they mean it just can't be shared, or if it doesn't exist. Either way, if I can get my dso203 loaded up and working then I'll be happy with that.

**Comment by MotoMaxis** [comment]
Here is a package of hex files (sys,spp,gpgs,loho) for older devices https://github.com/gabonator/LA104/files/6096611/DS203HW260-272.zip

**Comment by kneave** [comment]
Amazing, thank you! Can these be uploaded to the device by the UART connection too or do they need to be copied by the DFU drive please?

Overwise I assume I need to apply the DFU 3.10 file first, then the others in any order with a reboot of the device between?

**Comment by MotoMaxis** [comment]
Yes you can flash everything on DS203 using UART 3.3V TTL.  I believe DFU v3.45c works with Win 8 and above, however its only for hw v2.81 and v2.82 with 8mb build in storage. Your hw v2.60 I think has 2mb so won't work and you need older DFU.  If the DFU works, as in shows up on display I would leave it alone.  Flash the SYS, APP, FPGA and LOGO from https://github.com/gabonator/LA104/files/6096611/DS203HW260-272.zip and see if it boots.

Have a read here about the process.
https://forum.seeedstudio.com/t/tutorial-how-to-flash-debrick-update-dfu-on-dso203-hw-v2-81/18689
https://docs.google.com/document/d/1KtI75p771MsrvysjKE3TYNWSxzShYwZK9QapC75q7Is/edit

**Comment by kneave** [comment]
It looks like it's running now at least, though the 3.10 DFU didn't work and I had to use the 3.11 version. The mainboard has V2.60A on it so maybe that's actually 2.70 hardware or something in between? Either way, with 3.10 I couldn't see any devices when plugged in and in DFU mode, with 3.11 I can see STM Mass Storage in both Windows and Linux under lsusb.

If I run `fdisk -l` I can see the drive details too which is promising:
```
Disk /dev/sdd: 512 KiB, 524288 bytes, 1024 sectors
Disk model: DFU Disk
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
```

When I open parted it looks like the drive doesn't have a partition or anything on it though:

```
Error: /dev/sdd: unrecognised disk label
Model: Vertual DFU Disk (scsi)
Disk /dev/sdd: 524kB
Sector size (logical/physical): 512B/512B
Partition Table: unknown
Disk Flags:
```

I've seen another post reference "Vertual" rather than "virtual" so think I'm on the right track.

I adapted the command line from another issue on here and used the following for mine, I noticed the sector size on mine was reporting as 512 rather than 4096 on the other post:
'mkfs.msdos -s 1 -n DSO203 -f 2 -v -F 12 -S 512 -r 512 -I /dev/sdd'

I could then: 
'mkdir -p /media/usb'
'mount -t /dev/sdd /media/usb'

and it looks happier, but if I 'ls' in that folder, or even 'ls -lah' in case the '.RDY' file is hidden I can't see anything. I just had a look in gparted in the GUI and I can see the drive isn't formatted again either so maybe it didn't take?

<img width="279" alt="image" src="https://github.com/gabonator/LA104/assets/733411/115bb910-2187-4201-b0cc-82b1af03270a">

I guess it's also possible that I do in fact need the 3.10 DFU file and that the file I have is corrupt in some way?

I'm all a bit out of ideas and it's very late now though so I'm going to log off for the eve and hope I wake up with some fresh inspiration 😅

**Comment by kneave** [comment]
I've still not had any luck getting DFU working and for the life of my I can't find a backed up ISO for Windows 7. I dont suppose you had any luck with a full image with LA104 on it did you please?

**Comment by gramster** [comment]
> Here is a package of hex files (sys,spp,gpgs,loho) for older devices https://github.com/gabonator/LA104/files/6096611/DS203HW260-272.zip

What about the system.hex file? I'd like to try these on my 2.70 hardware.
